textattack/loggers/visdom_logger.py

```diff
@@ -16,9 +17,22 @@ def __init__(self, env='main', port=8097, hostname='localhost'):
         if not port_is_open(port, hostname=hostname):
             raise socket.error(f'Visdom not running on {hostname}:{port}')
         self.vis = Visdom(port=port, server=hostname, env=env)
+        self.vis_args = {'port': port, 'server': hostname, 'env': env}
         self.windows = {}
         self.sample_rows = []
 
+    def __getstate__(self):
+        # Temporarily save socket handle b/c we can't copy it
+        tmp = self.vis
+        self.vis = None
+        state = copy.deepcopy(self.__dict__)
+        self.vis = tmp
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        self.vis = Visdom(**self.vis_args)
```

Comment:
Consider removing self.vis_args and saving the other args as variables, then changing this line to 
`self.vis = Visdom(port=self.port, server=self.hostname, env=self.env)`

Reply:
Done